inductor: add device and dtype check before doing cpu fx packed weight

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99028

fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
inductor: add more check before doing cpu fx optimization

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add more check before doing cpu fx optimization"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add more check before doing cpu fx optimization"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add more check before doing cpu fx optimization"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add more check before doing cpu fx optimization"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add more check before doing cpu fx optimization"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on test/inductor/test_torchinductor.py:
```diff
@@ -1485,6 +1485,28 @@ def fn(a, b, c):
+        with torch.no_grad():
+            self.common(mod, (torch.randn(2, 8).to(torch.float64),))
+
+    # https://github.com/pytorch/pytorch/issues/96406
+    @requires_cuda()
+    def test_linear_mixed_device(self):
```
Maybe move to `test_cuda_repro.py`?

XiaobingSuper replied:
yes, moved.

XiaobingSuper committed:
Update on "inductor: add device and dtype check before doing cpu fx packed weight"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented on test/inductor/test_torchinductor.py:
```diff
@@ -1485,6 +1485,13 @@ def fn(a, b, c):
         )
 
+    # https://github.com/pytorch/pytorch/issues/98979
+    @unittest.skipIf(HAS_CUDA, "cuda failed for float64 linear")
```
I skip this case for cuda path which it is always failed.  @jansel @desertfire @ngimel

XiaobingSuper committed:
Update on "inductor: add device and dtype check before doing cpu fx packed weight"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: add device and dtype check before doing cpu fx packed weight"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot rebase

XiaobingSuper commented:
@pytorchbot rebase --branch=main

XiaobingSuper committed:
Update on "inductor: add device and dtype check before doing cpu fx packed weight"


fix https://github.com/pytorch/pytorch/issues/96406 and https://github.com/pytorch/pytorch/issues/98979

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge -f "test env issue, not related to this PR"
